tests/SpecTests/TransactionsSpecTest.php

```diff
@@ -36,6 +37,9 @@ class TransactionsSpecTest extends FunctionalTestCase
     private static $incompleteTests = [
         'error-labels: add unknown commit label to MaxTimeMSExpired' => 'PHPC-1382',
         'error-labels: add unknown commit label to writeConcernError MaxTimeMSExpired' => 'PHPC-1382',
+        'mongos-recovery-token: commitTransaction retry fails on new mongos' => 'isMaster failpoints cannot be disabled',
```

Comment:
I was not able to reliably clear the `failCommand` fail point if it's configured to fail a certain amounts of times. It seems that we don't trigger the fail point the expected number of times, which causes issues when removing the fail point (as `isMaster` will cause an error). However, even trying to remove the fail point in a loop did not always clear the fail point, which causes failures in following tests.

Reply:
Reading the [script for this test](https://github.com/mongodb/specifications/blob/c08e1be322477aaefb8d42f8d50e30fbc15cc1d0/source/transactions/tests/mongos-recovery-token.yml#L177) and the comments therein, I wonder if it was written with multi-threaded SDAM in mind. I wasn't aware we had any tests failing `isMaster` commands before reading this.

It looks like this test dates back to the original spec change for recovery tokens in [SPEC-1168](https://jira.mongodb.org/browse/SPEC-1168).

Perhaps you can reach out to @bazile-clyde to see whether the C driver is able to test this with a multi- or single-threaded client. Multi-threaded clients would offload all `isMaster` activity to a separate background thread and socket, which means these `isMaster` failures might not affect the client under test at all.

Reply:
Ah, the comments aren't present in the JSON files, which is why I missed the detailed explanation. The test is implemented in the C driver, but I couldn't find out whether it uses a single- or multi-threaded client. I'll check with Clyde on this.

Reply:
All of our fail points are activated using single-threaded clients. Also, to disable the fail point we didn't use any loops but instead just sent the command `{'configureFailPoint': 'failCommand' 'mode': 'off'}`. Both activating and deactivating fail points were done via a new client, pinned to the same server ID as the one we used for testing.